### PR TITLE
feat: implement dht values validators

### PR DESCRIPTION
**Pull request by SweeXordious**
## Overview

Closes https://github.com/celestiaorg/orchestrator-relayer/issues/72 and https://github.com/celestiaorg/orchestrator-relayer/issues/73

## Checklist

- [ ] New and updated code has appropriate documentation
- [ ] New and updated code has new and/or updated testing
- [ ] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [ ] Linked issues closed with keywords

### CODE REVIEW

File: p2p/validators.go [hunk 1]
```diff
@@ -1,35 +1,145 @@
...
+
+	// check if the evm address in the key is the same as the one in the confirm
+	if !strings.EqualFold(vsc.EthAddress, evmAddr) {
+		return ErrNotTheSameEVMAddress
+	}
+
+	// check if the signature is a valid signature.
+	// uses SigToVRS because it also does validation on the signature before returning the result.
+	// it would be better to check if the signature corresponds to the address. However, we don't
+	// have access to the digest that was signed in this level.
```
Comment (SweeXordious) [inline p2p/validators.go:48]: This will be added in a separate PR

Comment (SweeXordious) [reply p2p/validators.go:48]: https://github.com/celestiaorg/orchestrator-relayer/issues/224

**Review by evan-forbes — COMMENTED** [review]
just to clarify for my own understanding, we don't need to do a stateful check (check that the evm address is one that is expected), because we're we're not querying for keys unless they are from the validator set, correct?

**Comment by SweeXordious** [comment]
> check that the evm address is one that is expected

There is no way to know that the EVM address is one that is expected from the P2P network perspective

> we're we're not querying for keys unless they are from the validator set

anyone can post faulty signatures (if they tweak the orchestrator-relayer software), but a relayer will always query the correct thing that will be 100% accepted by the QGB contract. Because the same checks done on the QGB contract are done in these checks (including implementing this https://github.com/celestiaorg/orchestrator-relayer/issues/224)

**Review by evan-forbes — APPROVED** [review]
cool approving now and I think it makes sense, but I have some questions that are best dicussed syncronously or by code diving into libp2p
